wcivf/apps/elections/import_helpers.py

```diff
@@ -91,11 +91,10 @@ def __init__(self, ee_helper=None):
         self.post_cache = {}
 
     def update_or_create_from_ballot_dict(self, ballot_dict):
-        created = False
         # fall back to slug here as some temp ballots don't have an ID set
-        post_id = ballot_dict["post"].get("id", ballot_dict["post"]["slug"])
-        if not post_id in self.post_cache:
-            post, created = Post.objects.update_or_create(
+        post_id = ballot_dict["post"]["id"] or ballot_dict["post"]["slug"]
```

Comment:
Good spot!

I added this [recently](https://github.com/DemocracyClub/WhoCanIVoteFor/commit/d4541719ad013a21d24532f033e4a0ad33a7a2a5) as a response to some posts getting assigned to the incorrect ballot - this was because `self.post_cache` contained None values for >1 post.

This fix makes the code correct here, but I wanted to check that we didn't have duplicate slugs in YNR for some reason. The good news is that all posts without an identifier have a unique slug there, so this change should work fine :+1:

Reply:
Would it make sense to update the `ynr_id` to remove `blank=True` and add `unique=True`? It is already flagged as the primary key

Reply:
Removing `black=True` is a good idea - I'm not sure why it's there!

When a field is marked as the [primary key it automatically gets](https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.Field.primary_key) `unique=True` and `db_index=True`, so no need to add them as well.